tsinfer/inference.py

```diff
@@ -1391,7 +1410,36 @@ def _find_path(self, child_id, haplotype, start, end, thread_index=0):
         match = self.match[thread_index]
         missing = haplotype == tskit.MISSING_DATA
 
-        left, right, parent = matcher.find_path(haplotype, start, end, match)
+        key = f"{child_id}_{self.__class__.__name__}".encode()
+
+        if self.resume_lmdb is not None:
+            with self.resume_lmdb.begin() as txn:
```

Comment:
I'm having second thoughts about doing this in the worker threads. Although this is the place to put the dB lookup to minimise code changes, it introduces a db lookup in each worker. This will be hard to test, and introduce mutexes (if the dB does concurrent updates properly). Can we do the lookup in the main process, even if it means a bit of duplication between the samples and ancestors code paths?

This should also simplify doing it under dask

Reply:
Yes, `lmdb` actually uses a file-based lock so not ideal. I'll try to think of a way that this can be abstracted to work for both dask and local paths.